web/src/test/java/com/replica/StreetEdgeExporterTest.java

```diff
@@ -62,34 +60,80 @@ public static void setUp() {
             throw new IllegalStateException(TRANSIT_DATA_DIR + " directory should not already exist.");
         }
         Preconditions.checkState(transitDataDir.mkdir(), "could not create directory " + TRANSIT_DATA_DIR);
-    }
 
-    @AfterAll
-    public static void cleanUp() {
-        Helper.removeDir(new File(TARGET_DIR));
-        Helper.removeDir(new File(TRANSIT_DATA_DIR));
-    }
-
-    @Test
-    public void testExportEndToEnd() throws IOException {
         // Setup necessary mock
         final JarLocation location = mock(JarLocation.class);
         when(location.getVersion()).thenReturn(Optional.of("1.0.0"));
 
         // Add commands you want to test
-        final Bootstrap<GraphHopperServerConfiguration> bootstrap = new Bootstrap<>(new GraphHopperApplication());
+        bootstrap = new Bootstrap<>(new GraphHopperApplication());
         bootstrap.addBundle(new GraphHopperBundle());
         bootstrap.addCommand(new ImportCommand());
         bootstrap.addCommand(new ExportCommand());
 
         // Build what'll run the command and interpret arguments
-        Cli cli = new Cli(location, bootstrap, System.out, System.err);
+        cli = new Cli(location, bootstrap, System.out, System.err);
         cli.run("import", "test-data/beatty-sample-feed-config-car.yml");
+    }
+
+    @AfterAll
+    public static void cleanUp() {
+        Helper.removeDir(new File(TARGET_DIR));
+        Helper.removeDir(new File(TRANSIT_DATA_DIR));
+    }
+
+    @Test
+    public void testExportEndToEnd() throws IOException {
         cli.run("export", "test-data/beatty-sample-feed-config-car.yml");
         CSVFormat format = StreetEdgeExporter.CSV_FORMAT;
         File expectedOutputLocation = new File(TARGET_DIR + "street_edges.csv");
         CSVParser parser = CSVParser.parse(expectedOutputLocation, StandardCharsets.UTF_8, format);
         List<CSVRecord> records = parser.getRecords();
         assertEquals(769, records.size());
     }
+
+    @Test
+    public void testExportSingleRecord() {
+        GraphHopperServerConfiguration config = new GraphHopperServerConfiguration();
+        config.getGraphHopperConfiguration().
+                putObject("graph.flag_encoders", "car,foot").
+                putObject("datareader.file", "test-data/beatty.osm").
+                putObject("gtfs.file", "test-data/sample-feed.zip").
+                putObject("graph.location", TARGET_DIR).
+                setProfiles(Collections.singletonList(new Profile("foot").setVehicle("foot").setWeighting("fastest")));
+
+        // TODO copied from ExportCommand
```

Comment:
I don't know if there's a succinct way to extract / share this logic that exists in `ExportCommand`; the lines below in particular generate 5 `Map`s that we use as input to the function below.